apps/api/src/server/applications/s51advice/s51-advice.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * NSIP S51 Advice
+ *
+ * @typedef {Object} NsipS51AdvicePayload
+ * @property {number} adviceId
+ * @property {number} caseId
+ * @property {string} title
+ * @property {string} from
+ * @property {string} agent
+ * @property {'phone' | 'email' | 'meeting' | 'post'} method
+ * @property {string} enquiryDate
+ * @property {string} enquiryDetails
+ * @property {string} adviceGivenBy
+ * @property {string} adviceDate
+ * @property {string} adviceDetails
+ * @property {'checked' | 'unchecked' | 'readytopublish' | 'published' | 'donotpublish'} status
+ * @property {'unredacted' | 'redacted'} redactionStatus
+ * @property {string[]} attachmentIds
+ * */
+
+/**
+ * @param {import('apps/api/src/database/schema.js').S51Advice} s51Advice
+ * @returns {NsipS51AdvicePayload}
+ * */
+export const buildNsipS51AdvicePayload = (s51Advice) => ({
+	adviceId: s51Advice.id,
+	caseId: s51Advice.caseId,
+	title: s51Advice.title,
+	from: s51Advice.enquirer,
+	agent: `${s51Advice.firstName} ${s51Advice.lastName}`,
+	method: s51Advice.enquiryMethod,
+	enquiryDate: s51Advice.enquiryDate,
+	enquiryDetails: s51Advice.enquiryDetails,
+	adviceGivenBy: s51Advice.adviser,
+	adviceDate: s51Advice.adviceDate,
+	adviceDetails: s51Advice.adviceDetails,
+	status: s51Advice.publishedStatus,
+	redactionStatus: s51Advice.redactedStatus,
+	// @ts-ignore
+	attachmentIds: s51Advice.S51AdviceDocument.map((doc) => doc.id)
```

Comment:
Can probs remove lots of this boilerplate with pick

Reply:
Good shout 👍